src/RequiredCheckbox.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from "react"
+import './RequiredCheckbox.css';
+
+class RequiredCheckbox extends Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {
+      changed: false,
+      checked: this.props.checked || false,
+    }
```

Comment:
you can set an initial `state` directly on the class and bind that `handleChange` to the instance by using class prop notation, which looks like
```js
class RequiredCheckbox extends Component {
  state = {
    changed: false,
    checked: this.props.checked || false
  }

  handleChange = e => {
    ...
  }
}
```